internal/docker/container.go

```diff
@@ -309,6 +313,61 @@ func (r *ContainerRunner) collectResults(results chan result, expected int) bool
 	return passed
 }
 
+func (r *ContainerRunner) downloadArtifacts(artifactsCfg config.ArtifactDownload, res result) {
```

Comment:
Isn't this (and all of the related code here) a duplicate of the sauce cloud implementation? Is there way we can avoid code duplication in this case?

Reply:
Thinking 🤔